homeassistant/components/proxmoxve/sensor.py

```diff
@@ -0,0 +1,195 @@
+        self._item_node = item_node
+        self._item_type = item_type
+        self._item_id = item_id
+        self._name = None
+        self._proxmox_client = proxmox_client
+        self._sensor_type = sensor_type
+        self._state = None
+        self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
+
+    @property
+    def name(self):
+        """Return the name of the entity."""
+        return self._name
+
+    @property
+    def device_class(self):
+        """Return the class of this sensor."""
+        return self._device_class
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return self._icon
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return self._unit_of_measurement
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return self._available
+
+    def update(self):
+        """Update sensore values."""
+        items = (
+            self._proxmox_client.get_api_client()
+            .nodes(self._item_node)
+            .get(self._item_type.name)
+        )
+        item = next(
+            (item for item in items if item["vmid"] == str(self._item_id)), None
+        )
+
+        if item is None:
+            _LOGGER.warning("Couldn't find VM/Container with the ID %s", self._item_id)
+            return None
+
+        self._name = (
+            f"{self._item_node} {item['name']} {SENSOR_TYPES[self._sensor_type][0]}"
+        )
+
+        item["disk"] = int(item["disk"])
+        item["maxdisk"] = int(item["maxdisk"])
+
+        if self._sensor_type == "cpu":
+            self._state = item["cpus"]
+        elif self._sensor_type == "cpu_use_percent":
+            self._state = round(item["cpu"] * 100, 1)
+        elif self._sensor_type == "mem":
+            self._state = round(item["maxmem"] / (1024 ** 2), 1)
+        elif self._sensor_type == "memory_free":
+            self._state = round((item["maxmem"] - item["mem"]) / (1024 ** 2), 1)
+        elif self._sensor_type == "memory_use":
+            self._state = round(item["mem"] / (1024 ** 2), 1)
+        elif self._sensor_type == "memory_use_percent":
+            self._state = round(item["mem"] / item["maxmem"] * 100, 1)
+        elif self._sensor_type == "disk":
+            self._state = round(item["maxdisk"] / (1024 ** 3), 1)
+        elif self._sensor_type == "disk_free":
+            if self._item_type == ProxmoxItemType.lxc:
+                self._state = round((item["maxdisk"] - item["disk"]) / (1024 ** 3), 0)
+        elif self._sensor_type == "disk_use":
+            if self._item_type == ProxmoxItemType.lxc:
+                self._state = round(item["disk"] / (1024 ** 3), 1)
+        elif self._sensor_type == "disk_use_percent":
```

Comment:
All `elif`s can be changed to `if`, as there is no `else`.

Reply:
@bouwew you are right, that always only a one condition will match, but all other conditions are not needed to be evaluated after a match occur and therefore I would still prefer the `elif` approach.

Reply:
but I now i see, that the checks for `_item_type` are not needed, because allready handled in `setup_platform()` 😄

Reply:
This is now a moot point as we have transitioned to a cleaner (and more performant) system without all these elifs.